### ISSUE TITLE
Add cooling capacity field to racks

### ISSUE THREAD

**Issue by shield-9** [netbox-community/netbox#15006]
### NetBox version

v3.7.1

### Feature type

Change to existing functionality

### Proposed functionality

* Add "cooling capacity" (unit: watts) field to Racks
* Add "cooling utilization" bar graph to Rack details screen
    * Can be calculated as `{Allocated draw of devices in the rack} / {Cooling capacity of the rack}`

Depending on the cooling method and design, it may be possible to calculate the cooling capacity of each rack in some way for some environments, but I think it is difficult to generalize and should be achieved by plugin.

### Use case

All electronic devices consume power and also generate heat. In recent years, high-power devices such as GPU servers have been increasing in the market, and when installing such devices into racks, it is often necessary to consider not only the power consumption but also the amount of heat generated.

A rack's power and cooling capacity is constrained by different factors, such as:
* Power supply capacity: bus duct and UPS capacity, PDU specifications, etc.
* Cooling capacity: layout and specifications of air conditioners, CDUs, RDHx, chillers, other racks (nearby heat sources), etc.

So, to better understand rack utilization, it is good to have cooling utilization as an indicator, in addition to power utilization.

**Comment by alehaa** [comment]
In my opinion, it would make sense to add this to locations as well, since smaller locations might not be cooling per rack, but per room / row / etc.

**Comment by goteamkor** [comment]
How would you model a cooling utilization without also having to model air conditioners?
If you have all these devices that are generating heat from electricity, you also need to have something on the other side that is spitting out cold air to be able to mathematically determine a percentage of capacity available/used.

It would be cool for NetBox to automatically calculate BTU's based on the maximum and/or allocated draw (in watts) of power ports.
Then you might need an additional model for HVAC so you can also model how much cooling capacity is available.

**Comment by alehaa** [comment]
> The thing is, rack's don't have a cooling capacity.

I think there are racks available with integrated cooling, fire supression, etc.

Maybe this could be handled easily by adding cooling capacity to devices. Then one could make a `btu` property, which is either positive and calculated from power consumption, or negative as defined in cooling capacity. The sum of `btu` fields in a rack / location should be negative (i.e. more cooling capacity then heat is produced).

**Comment by goteamkor** [comment]
> I think there are racks available with integrated cooling, fire supression, etc.

true, these come to mind: https://www.apc.com/us/en/product-range/61780-uniflair-direct-expansion-inrow-cooling/#products
I agree on adding cooling capacity to devices. Then you can model both the above rack coolers, as well as an actual HVAC device.
